lib/icingadb/redisconnection.cpp

```diff
@@ -500,3 +504,14 @@ void RedisConnection::WriteOne(RedisConnection::Query& query, asio::yield_contex
 		WriteOne(m_UnixConn, query, yc);
 	}
 }
+
+/**
+ * Specify a callback that is run each time a connection is successfully established
+ *
+ * The callback is executed from a Boost.Asio coroutine and should therefore not perform blocking operations.
+ *
+ * @param callback Callback to execute
+ */
+void RedisConnection::SetConnectedCallback(const std::function<void(asio::yield_context& yc)>& callback) {
+	m_ConnectedCallback = callback;
+}
```

Comment:
Why don’t you omit `const&` and move the param?

Reply:
Is there an advantage in doing so? One copies from a reference, the other copies and then moves.

Reply:
The other _constructs_ and then moves, doesn’t it?

Reply:
And this one?

Reply:
You mean my original version? That one constructs a `std::function`, passes it as a reference and then copy assigns it to the class member.